derotation/analysis/incremental_rotation_pipeline.py

```diff
@@ -156,6 +154,29 @@ def get_interpolated_angles(self) -> np.ndarray:
 
         return interpolated_angles * -1
 
+    def check_rotation_number_after_interpolation(
```

Comment:
this is quite similar to `full_rotation_pipeline.check_rotation_number_after_interpolation()`, I think (?) the only difference is `self.number_of_rotations` vs `1` on the final check. Maybe these could be combined if it makes sense to?

Reply:
I want them to be explicitly separated as the number of rotations is `>1` in one experimental protocol and exactly `1` in the other